session/session.go

```diff
@@ -3152,10 +3055,8 @@ func (s *session) PrepareTSFuture(ctx context.Context, future oracle.Future, sco
 	return nil
 }
 
+// GetPreparedTxnFuture returns the TxnFuture
 func (s *session) GetPreparedTxnFuture() sessionctx.TxnFuture {
-	if !s.txn.validOrPending() {
```

Comment:
Why remove this? I think it is reasonable if a txn is not prepared the `TxnFuture` should return a nil value.